### ISSUE TITLE
When globalName is set to something other than "nuxt", there is an Uncaught TypeError

### ISSUE THREAD

**Issue by sam3d** [nuxt/nuxt.js#4296]
### Version

[v2.2.0](https://github.com/nuxt.js/releases/tag/v2.2.0)

### Reproduction link

[https://codesandbox.io/s/430mvo2zo0](https://codesandbox.io/s/430mvo2zo0)

### Steps to reproduce

1. Create the most basic `nuxt.js` app that can be created (with a single `pages/index.vue`)
2. In the `nuxt.config.js` set the following:

```javascript
export default {
    globalName: "root"
}
```

3. The web console will now throw the following error:

```
VM63:1 Uncaught TypeError: Cannot read property 'constructor' of undefined
    at setTimeout (<anonymous>:1:123)
```

### What is expected ?

No error is thrown when a different `globalName` is set

### What is actually happening?

The TypeError (shown above) is thrown

### Additional comments?

This happens on any `globalName` that isn't `nuxt`

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8110">#c8110</a>)</em></sub></div>

**Comment by sam3d** [comment]
Ah, okay. This appears to happen on interaction with Vue DevTools, otherwise it doesn't occur. It likely isn't an error with Nuxt.

**Comment by galvez** [comment]
Thanks guys, I'll take a look at this tonight.

**Comment by galvez** [comment]
Fixed via #4305
